File: Lab4/Invert.java
public class Invert 
{
    public static TreeNode invertTree(TreeNode root)
    {
        TreeNode temp;
        if(root.left != null & root.right != null)
        {
            temp = root.left;
            root.left = root.right;
            root.right = temp;
            invertTree(root.left);
            invertTree(root.right);
        }
        return root;
    }
    public static void main(String[] args)
    {
        TreeNode root = new TreeNode(4);
            TreeNode three = root.left = new TreeNode(3);
            TreeNode eight = root.right = new TreeNode(8);
            TreeNode one = root.left.left = new TreeNode(1);
            TreeNode ten = root.left.right = new TreeNode(10);
            TreeNode five = root.right.left = new TreeNode(5);
            TreeNode nine = root.right.right = new TreeNode(9);

        TreeNode holder = invertTree(root);
        System.out.println(holder.val);
        System.out.println(holder.left.val + " " + holder.left.left.val + " " + holder.left.right.val);
        System.out.println(holder.right.val + " " + holder.right.left.val + " " + holder.right.right.val);
    }
}

File: Lab4/Invert.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def invertTree(root):
    if root and root.left and root.right:
        temp = root.left
        root.left = root.right
        root.right = temp
        invertTree(root.left)
        invertTree(root.right)
    return root

if __name__ == "__main__":
    root = TreeNode(4)
    root.left = TreeNode(3)
    root.right = TreeNode(8)
    root.left.left = TreeNode(1)
    root.left.right = TreeNode(10)
    root.right.left = TreeNode(5)
    root.right.right = TreeNode(9)

    holder = invertTree(root)
    print(holder.val)
    print(holder.left.val, holder.left.left.val, holder.left.right.val)
    print(holder.right.val, holder.right.left.val, holder.right.right.val)
